lnst/Controller/MachineMapper.py

```diff
@@ -345,3 +345,57 @@ def get_mapping(self):
         if self._virtual_matching:
             mapping["virtual"] = True
         return mapping
+
+
+class ContainerMapper(object):
+    """Implements simple matching algorithm that maps containers to requirements.
+    Containers are created in `ContainerPoolManager` using requirements.
+
+    Example runner:
+
+    recipe = Recipe()
+    ctl = Controller(poolMgr=ContainerPoolManager, mapper=ContainerMapper("unix:///run/podman/podman.sock", "lnst"))
+    ctl.run(recipe)
+    """
+    def __init__(self, podman_uri: str, image: str):
+        super().__init__()
+
+        self._podman_url = podman_uri
+        self._image = image
+
+        self._pool_manager = None
+        self._mreqs = {}
+
+    def set_requirements(self, mreqs: dict):
+        self._pool_manager.podman_connect(self._podman_url)
+        self._pool_manager.image = self._image
+        self._pool_manager.process_reqs(mreqs)
+
+        self._mreqs = mreqs
+
+    def set_pools(self, pool_manager):
+        """`ContainerMapper` does not support multiple pools but it requires pool manager.
+        """
+        self._pool_manager = pool_manager
+
+    def matches(self):
+        """1:1 mapping of containers to requirements
+        """
+        mapping = {"machines": {}, "networks": {}, "virtual": False, "pool_name": "default"}
+        pool = self._pool_manager.get_pool()
+
+        for m_id, reqs in self._mreqs.items():
+            hostname = pool[m_id]["params"]["hostname"]
+            machine = mapping["machines"][m_id] = {"target": m_id, "hostname": hostname, "interfaces": {}}
+
+            for inf_id, infs in reqs["interfaces"].items():
+                network_name = self._pool_manager.get_network_name(infs["network"])
+
+                (machine_inf_id, machine_inf), = filter(lambda pair: pair[1]["network"] == network_name, pool[m_id]["interfaces"].items())
```

Comment:
As we found out at "offline" discussion, it would be pretty complicated to implement support for connecting 2 nics of 1 container to a single network.  
For now, I left it unsupported.